Makefile.e2e.ci

```diff
@@ -134,3 +134,58 @@ manual-rotate-prober-sa-key: __download_prober_creds
 # Manually rotate the prober's service account key in OSS prow.
 manual-rotate-prober-sa-key-oss: __download_prober_creds
 	GCP_PROBER_CREDS=$(SA_KEY_FILE) ./scripts/oss-prow/prober-runner-sa-key-rotate.sh
+
+# The CI target that runs the first group of tests with the multi-repo mode on GKE cluster.
+test-e2e-gke-multi-repo-test-group1:
+	$(MAKE) E2E_ARGS="$(E2E_ARGS) --share-test-env --multirepo --test-features=acm-controller,nomos-cli,override-api" \
+		test-e2e-go-gke-ci
+
+# The CI target that runs the second group of tests with the multi-repo mode on GKE cluster.
+test-e2e-gke-multi-repo-test-group2:
+	$(MAKE) E2E_ARGS="$(E2E_ARGS) --share-test-env --multirepo" --test-features=cluster-selector,hydration" \
+		test-e2e-go-gke-ci
+
+# The CI target that runs the third group of tests with the multi-repo mode on GKE cluster.
+test-e2e-gke-multi-repo-test-group3:
+	$(MAKE) E2E_ARGS="$(E2E_ARGS) --share-test-env --multirepo --test-features=drift-control,lifecycle" \
+		test-e2e-go-gke-ci
+
+# The CI target that runs the fourth group of tests with the multi-repo mode on GKE cluster.
+test-e2e-gke-multi-repo-test-group4:
+	$(MAKE) E2E_ARGS="$(E2E_ARGS) --share-test-env --multirepo --test-features=multi-repos" \
+		test-e2e-go-gke-ci
+
+# The CI target that runs the fifth group of tests with the multi-repo mode on GKE cluster.
+test-e2e-gke-multi-repo-test-group5:
+	$(MAKE) E2E_ARGS="$(E2E_ARGS) --share-test-env --multirepo --test-features=reconciliation1" \
```

Comment:
Can we use an environment variable somehow to track the features to test? If we do that, we won't need these `test-e2e-gke-multi-repo-test-groupXX` make rules.

Reply:
Yes, we can. But then we will configure the env variable in the Prow job. So the grouping logic is outside the project. 

The purpose of creating separate targets explicitly here is to keep the grouping logic inside the `kpt-config-sync` project and the prow jobs only need to invoke corresponding targets.

It will be easier to rebalance the testing groups within the same project in the future.

Reply:
With the multiple similar but slightly different targets, we can have the following benefits:
- Keep the grouping logic self-contained within the kpt-config-sync project.
- It is easier to run a particular test group locally by invoking the target directly, instead of specifying the features.
- It is in line with the nomos-operator settings. The operator has multiple similar targets for different platforms.

For presubmit jobs, we can have separate targets.